TombEngine/Game/control/box.cpp

```diff
@@ -139,52 +140,34 @@ bool MoveCreature3DPos(Pose* fromPose, Pose* toPose, int velocity, short angleDi
 
 void CreatureYRot2(Pose* fromPose, short angle, short angleAdd) 
 {
-	if (angleAdd < angle)
-	{
+	if (angle > angleAdd)
 		fromPose->Orientation.y += angleAdd;
-		return;
-	} 
-
-	if (angle < -angleAdd)
-	{
+	else if (angle < -angleAdd)
 		fromPose->Orientation.y -= angleAdd;
-		return;
-	} 
-
-	fromPose->Orientation.y += angle;
+	else
+		fromPose->Orientation.y += angle;
 }
 
 bool SameZone(CreatureInfo* creature, ItemInfo* target)
 {
-	int* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
-	auto* item = &g_Level.Items[creature->ItemNumber];
-
-	auto* room = &g_Level.Rooms[item->RoomNumber];
-	FloorInfo* floor = GetSector(room, item->Pose.Position.x - room->x, item->Pose.Position.z - room->z);
-	if (floor->Box == NO_BOX)
-		return false;
-
-	item->BoxNumber = floor->Box;
-
-	room = &g_Level.Rooms[target->RoomNumber];
-	floor = GetSector(room, target->Pose.Position.x - room->x, target->Pose.Position.z - room->z);
-
-	if (floor->Box == NO_BOX)
-		return false;
+	auto zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
+	auto item = &g_Level.Items[creature->ItemNumber];
 
-	target->BoxNumber = floor->Box;
+	auto r1 = &g_Level.Rooms[item->RoomNumber];
+	item->BoxNumber = GetSector(r1, item->Pose.Position.x - r1->z, item->Pose.Position.z - r1->z)->Box;
+	auto r2 = &g_Level.Rooms[target->RoomNumber];
+	target->BoxNumber = GetSector(r2, target->Pose.Position.x - r2->z, target->Pose.Position.z - r2->z)->Box;
 
-	return (zone[item->BoxNumber] == zone[target->BoxNumber]);
+	return zone[item->BoxNumber] == zone[target->BoxNumber];
 }
 
 short AIGuard(CreatureInfo* creature) 
 {
-	auto* item = &g_Level.Items[creature->ItemNumber];
-	if (item->AIBits & (GUARD | PATROL1))
+	auto item = &g_Level.Items[creature->ItemNumber];
+	if (item->AIBits & MODIFY)
 		return 0;
 
-	int random = GetRandomControl();
-
+	int random = Random::GenerateInt();
```

Comment:
`TestProbability` should be used instead. Moreover, `GetRandomControl` returned short, while `GenerateInt` returns integer, therefore probability ratio will be different here. Calculate it based on assumption of `GetRandomControl` range.